env/lib/python3.12/site-packages/pip/_internal/cli/cmdoptions.py

```
@@ -0,0 +1,1074 @@
+
+    # Installations or downloads using dist restrictions must not combine
+    # source distributions and dist-specific wheels, as they are not
+    # guaranteed to be locally compatible.
+    if dist_restriction_set and sdist_dependencies_allowed:
+        raise CommandError(
+            "When restricting platform and interpreter constraints using "
+            "--python-version, --platform, --abi, or --implementation, "
+            "either --no-deps must be set, or --only-binary=:all: must be "
+            "set and --no-binary must not be set (or must be set to "
+            ":none:)."
+        )
```

Ruff (0.13.3)

Avoid specifying long messages outside the exception class

(TRY003)